lib/astutils.cpp

```diff
@@ -3647,3 +3647,87 @@ bool isUnevaluated(const Token *tok)
 {
     return Token::Match(tok, "alignof|_Alignof|_alignof|__alignof|__alignof__|decltype|offsetof|sizeof|typeid|typeof|__typeof__ (");
 }
+
+static std::set<std::string> getSwitchExprs(const Token *startbrace, bool &hasDefault)
+{
+    std::set<std::string> exprs;
+    const Token *endbrace = startbrace->link();
+    if (!endbrace)
+        return {};
+
+    int braces = 0;
+    hasDefault = false;
+    for (const Token *tok = startbrace->next(); tok && tok != endbrace; tok = tok->next()) {
+        if (Token::simpleMatch(tok, "{")) {
```

Comment:
if you want to skip inner scopes it's better with `tok = tok->link()` and skip the braces variable